webdav_sync.sh

```diff
@@ -58,6 +59,39 @@ url_decode() {
 }
 
 
+allFiles=''
+currentDir=$remote_dir_name
+
+getFilesRec()
+{
+    currentDirAbs="$remote_host$remote_path_suffix$currentDir"
+
+
+    dirsAndFiles=$(curl --silent --user "$user":"$passwd" $currentDirAbs -X PROPFIND --data '<?xml version="1.0"?>
+    <a:propfind xmlns:a="DAV:">
+    <a:prop>
+    <a:resourcetype />
+    </a:prop>
+    </a:propfind>'  | grep 'href' | sed 's/<\/\?[^>]\+>//g' | sed 's/HTTP\/1.1 200 OK/\n/g' | sed "s/$escaped_remote_path_suffix//g" | grep -v '^$')
+
+    dirs=$(echo "$dirsAndFiles" | grep '/$')
+    files=$(echo "$dirsAndFiles" | grep -v '/$')
+
+
+    allFiles="$allFiles"$'\n'$(echo "$files" | grep '\.epub$')
```

Comment:
Why only `epub`? Technically, we have to be able to sync anything. No one know all use cases of webdav sync.

Reply:
It was just because my only use case is to sync my books, and syncing the calibre library comes with some metadata files I didn't want to have on my ereader. If you want to have it more generic, just remove the additional pipe :)